src/fold/loop/checks.py

```diff
@@ -20,14 +16,6 @@ def check_types(
     else:
         assert isinstance(X, pd.DataFrame), "X must be a pandas DataFrame."
     assert isinstance(y, pd.Series), "y must be a pandas Series."
-
-    X_trimmed, _, _ = trim_initial_nans(X, y, Artifact.empty(X.index))
-    if len(X_trimmed) < len(X):
-        warn(
-            f"Detected initial NaNs in X ({ len(X) - len(X_trimmed) } instances),"
```

Comment:
this is now a separate step, where we actually trim. I'll copy the warning there!